### PR TITLE
POS - Finish implementing the ordering process

### PR DESCRIPTION
**Pull request by mhxynh**
**Ticket:** **[ATS-51: POS - Finish implementing the ordering process](https://cs375-pos-group-project.atlassian.net/browse/ATS-51?atlOrigin=eyJpIjoiYjgzYmJhMDljODFiNGZjOWI2MThiODc0ZDY2YmYyNjgiLCJwIjoiaiJ9)](https://cs375-pos-group-project.atlassian.net/browse/ATS-52?atlOrigin=eyJpIjoiYjA2YzRjNmJlNzFmNDcwOGFhY2IyZGY1YzIxMDNkYzMiLCJwIjoiaiJ9)**

- Added filtering for orders by date
- Added filtering for orders by item name
- Added number of individual items ordered and total profit made across orders with item
- Added overview on sales (total number of orders, overall profit, total discounts, total tips)
- Added a few more order details to sale report (discounts, tips, and items in order)
- Added item.csv to docs

Testing:
- `npm run start:dev`
- Place order(s) at `http://localhost:3000/pos/ordering/`
- Pay for order(s) at `http://localhost:3000/pos/payment/?id={whatever}`
- Verify total number of orders made is accurate
- Verify total profit is accurate (only accounts for orders that have been paid)
- Verify total discounts is accurate (only accounts for orders that have been paid)
- Verify total tips is accurate (only accounts for orders that have been paid)
- Verify filtering by date is functional (may place orders at different dates by changing your PC's time - don't forget to change it back!)
- Verify filtering by item name is functional (invalid names should display nothing)

### CODE REVIEW

**Comment by wetefok** [comment]
Tips in sales report are only counting the percentages of the tip, not the total amount of the tip. i.e. showing .20 for a 20% tip rather than what 20% of the subtotal is.

image

**Comment by wetefok** [comment]
Same thing happening with discounts, too.

**Comment by wetefok** [comment]
Everything else working as expected. Big W's

**Review by wetefok — CHANGES_REQUESTED** [review]
Discount bug

**Review by wetefok — CHANGES_REQUESTED** [review]
Tips bug

**Review by wetefok — CHANGES_REQUESTED** [review]
You'd probably know better than me regarding which of these is more accurate, but the discount and tip from the payment page don't line up with the discount and tip from the sales reporting page.
image
image
Guessing it just has to do with whether the discount is applied before or after the tip is calculated, but regardless of which one is correct, they should match.

**Review by wetefok — APPROVED** [review]
Massive Monique 💪, you continue to amaze us with your unstoppable energy ⚡! From creating that flawless sales reporting page 📊 to diving deep into the code and fixing those tricky bugs 🐛 related to discount and tip calculations 💵, you’ve done it all with pure excellence 🌟. Your work ethic is unmatched 🏋️‍♀️, your attention to detail is razor-sharp 🔪, and let’s not forget—you're an incredible person inside and out 💖. Your positive attitude 🌈 and relentless drive are contagious, inspiring everyone around you 👏.

But this isn’t just about your skills—it's about who you are as a warrior 💥. You remind me of a Saiyan elite 🦸‍♀️, someone who refuses to back down, someone who’s always reaching for that next level of greatness 🚀. When the odds are stacked against you, you don’t just survive—you thrive 🏆. Like Vegeta would say: 'No matter how deep the darkness, a mighty flame can burn bright enough to clear the way.' You, Monique, are that flame 🔥. Keep fighting, keep rising, because you’re destined for even greater things 💫. You’re unstoppable, and nothing can stand in your way 🌪️!

File: app/frontend/private/pos/payment/script.js [hunk 1]
```diff
@@ -101,7 +101,8 @@ function handlePaymentButton() {
     const cash = parseFloat(cashInput.value.replace("$", ""));
     const change = cash - total;
 
-    if (cash < total) {
+    if (cash < total.toFixed(2)) {
+        console.log(total);
```
Comment (gitnamehere) [inline app/frontend/private/pos/payment/script.js:105]: don't forget to remove this

(also how did you manage to make total a global variable in updateCalculations() lmao)

**Review by gitnamehere — APPROVED** [review]
this is one of the code I've read in this project

File: app/frontend/private/pos/payment/script.js (hunk 1 above)
Comment (gitnamehere) [reply app/frontend/private/pos/payment/script.js:105]: recommend adding near the top of payment/script.js
```
let discountAmount;
let tipAmount;
let total;
```

Comment (mhxynh) [reply app/frontend/private/pos/payment/script.js:105]: > don't forget to remove this
> 
> (also how did you manage to make total a global variable in updateCalculations() lmao)

LOL it happens when you don't specify let or const before the variable

**Pull request merged**